src/platform/posix/tcp_ip_channel.c

```diff
@@ -198,7 +210,7 @@ static lf_ret_t TcpIpChannel_check_socket_error(int fd) {
   }
 }
 
-static lf_ret_t TcpIpChannel_try_connect_client(NetworkChannel *untyped_self) {
+static lf_ret_t _try_connect_client(NetworkChannel *untyped_self) {
```

Comment:
the underscore is indicating that this is a "private" member function?

Reply:
Yes as far as I understood it the TcpIpChannel Prefix is to mark the functions that implement the interface, so all other internal code structuring functions are easier to differentiate with for example included functions by using underscore to mark them as private

Reply:
I am fine with this convention. But I would also be fine with omitting the `_`, since the functions are static they will not be visible outside this file anyways.

Reply:
For me the main advantage of it is to help differentiate between our own static functions and for example included library functions. I can directly see that for example `send_coap_message` is a private function and not a RIOT coap API function.

But I also don't have a strong opinion on it

Reply:
I mean within that c file of course

Reply:
how about:

`_<Class-Name>_<Member-Name>` for private member functions
`<Class-Name>_<Member-Name>` for public member functions

... would be a bit more consistent. Then the leading `_` would clearly indicate visibility.

Reply:
What I am not sure is would you consider function such as `TcpIpChannel_send_blocking` as private or as public?

From the code perspective they are `private`, but then exposed through the super type `NetworkChannel`.
Still different though compared to for example to `TcpIpChannel_ctor` which is indeed `public`.

So do we say:
- `TcpIpChannel_ctor` (public)
- `TcpIpChannel_send_blocking` (private, but exposed through super type)
- `_TcpIpChannel_some_internal_helper_function' (private)

Or:
- `TcpIpChannel_ctor` (public)
- `_TcpIpChannel_send_blocking` (private, but exposed through super type)
- `_TcpIpChannel_some_internal_helper_function' (private)

I personally think prefixing all the internal functions with `_TcpIpChannel_` makes the code somewhat harder to read, but I think it is still acceptable. But we need to make sure we are on the same page on how it is exactly implemented.

Reply:
I would say:

`TcpIpChannel_` means that it is a public facing API through function pointers on a struct, or through the header. `_TcpIpChannel_` means private helper functions

Reply:
Which is in accordance with the first

Reply:
If you agree, then could you update the naming so its in accordance with this, and then we can merge this?

Reply:
Yeah no problem

Reply:
Done ;)